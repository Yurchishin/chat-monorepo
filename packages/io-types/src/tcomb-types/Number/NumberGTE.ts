import t from 'tcomb-validation'
import { NumberGT } from './NumberGT'
import { Numerical } from './Numerical'

export const NumberGTE = (min: number) => t.union([
    Numerical(min),
    NumberGT(min),
], 'NumberGTE')
