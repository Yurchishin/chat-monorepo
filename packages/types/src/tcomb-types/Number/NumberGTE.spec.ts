import { NumberGTE } from './NumberGTE'
import { expectTypeSuccess, expectTypeFailure } from "../../tests"

describe('NumberGTE(5)', () => {
	const GTE5 = NumberGTE(5)

	expectTypeSuccess(GTE5, 6000)
	expectTypeSuccess(GTE5, 5)
	expectTypeFailure(GTE5, 4)
	expectTypeFailure(GTE5, '1')
})
