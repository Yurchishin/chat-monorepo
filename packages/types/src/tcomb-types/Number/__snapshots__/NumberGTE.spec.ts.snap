// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberGTE(5) expectTypeFailure(1): TYPE_FAILURE::NumberGTE(1) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "1",
      "expected": [Function],
      "message": "Invalid value \\"1\\" supplied to NumberGTE",
      "path": Array [],
    },
  ],
  "value": "1",
}
`;

exports[`NumberGTE(5) expectTypeFailure(4): TYPE_FAILURE::NumberGTE(4) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 4,
      "expected": [Function],
      "message": "Invalid value 4 supplied to NumberGTE",
      "path": Array [],
    },
  ],
  "value": 4,
}
`;
